{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read in processed RNA-seq, save into pickle."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GPL17021.soft',\n",
       " 'SRP061902',\n",
       " 'GPL13112.soft',\n",
       " 'GSE67835',\n",
       " 'GSE71585-GPL17021_series_matrix.txt',\n",
       " 'GSE52564',\n",
       " 'GPL15520.soft',\n",
       " 'GSE67835-GPL18573_series_matrix.txt',\n",
       " 'GSE71585-GPL13112_series_matrix.txt',\n",
       " 'SRP033200',\n",
       " 'GPL18573.soft',\n",
       " 'GSE71585',\n",
       " 'pickle',\n",
       " 'GSE73721-GPL18573_series_matrix.txt',\n",
       " 'GPL16417.soft',\n",
       " 'GSE60361',\n",
       " 'GSE52564_series_matrix.txt',\n",
       " 'GSE71585-GPL16417_series_matrix.txt',\n",
       " 'GSE60361_series_matrix.txt',\n",
       " 'GSE67835-GPL15520_series_matrix.txt',\n",
       " 'GSE73721-GPL19057_series_matrix.txt',\n",
       " 'SRP064454',\n",
       " 'SRR1033783_2.fastq.gz',\n",
       " 'GSE73721',\n",
       " 'SRP057196',\n",
       " 'GPL19057.soft',\n",
       " 'SRP045452']"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "root_directory = '/cellar/users/hsher/Data/ideker/sc/'\n",
    "import os\n",
    "os.listdir(root_directory)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# start from raw data.\n",
    "1. step one: untar/gunzip all files (manually)\n",
    "2. step two: read in all data into scanpy\n",
    "3. follow single cell protocol\n",
    "- filter read per cell\n",
    "- filter high mitochondrial content\n",
    "- normalize\n",
    "4. concat into the same anndata per species\n",
    "- visualize to check\n",
    "- calculate cluster (Adjusted Rand Index)\n",
    "5. map mouse genome to human and revisualize per subspace\n",
    "- are they comparable\n",
    "\n",
    "output: scanpy: anndata\n",
    "\n",
    "problems: \n",
    "1. different species problem\n",
    "- first seperate two data from human and mice\n",
    "    - combining data from different dataset simply by normalization will not eliminate batch effect? (wonder why paper do it?\n",
    "\n",
    "validation:\n",
    "- visualize cell cluster: is the clustering consistent with data\n",
    "    - check batch effect: see if there's clustering per dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Reading Dataset!\n",
    "## Dataset 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GSM1269919_WC3.xls',\n",
       " 'GSM1269903_Astrocyte1.xls',\n",
       " 'GSM1269918_WC2.xls',\n",
       " 'GSM1269910_NFO2.xls',\n",
       " 'GSM1269907_OPC1.xls',\n",
       " 'GSM1269906_Neuron2.xls',\n",
       " 'GSM1269905_Neuron1.xls',\n",
       " 'GSM1269916_Endothelial2.xls',\n",
       " 'GSM1269913_Microglia1.xls',\n",
       " 'GSM1269917_WC1.xls',\n",
       " 'GSM1269915_Endothelial1.xls',\n",
       " 'GSM1269914_Microglia2.xls',\n",
       " 'GSM1269912_MO2.xls',\n",
       " 'GSM1269909_NFO1.xls',\n",
       " 'GSM1269908_OPC2.xls',\n",
       " 'GSM1269911_MO1.xls',\n",
       " 'GSM1269904_Astrocyte2.xls']"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# process /GSE52564\n",
    "data_1 = root_directory + 'GSE52564/'\n",
    "filelist = os.listdir(data_1)\n",
    "filelist = [f for f in filelist if 'xls' in f]\n",
    "filelist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "dflist = []\n",
    "for f in filelist:\n",
    "    df = pd.read_excel(data_1+f, index_col = 0) # read into dataframe\n",
    "    dflist.append(df)\n",
    "all_df = pd.concat(dflist, axis = 1, ignore_index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>WB3</th>\n",
       "      <th>A1</th>\n",
       "      <th>WB2</th>\n",
       "      <th>GC2</th>\n",
       "      <th>OPC1</th>\n",
       "      <th>N2</th>\n",
       "      <th>N1</th>\n",
       "      <th>Endo2</th>\n",
       "      <th>MGL1</th>\n",
       "      <th>WB1</th>\n",
       "      <th>Endo1</th>\n",
       "      <th>MGL2</th>\n",
       "      <th>MOG2</th>\n",
       "      <th>GC1</th>\n",
       "      <th>OPC2</th>\n",
       "      <th>MOG1</th>\n",
       "      <th>A2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene.symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0610005C13Rik</th>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.200201</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "      <td>0.100000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610007C21Rik</th>\n",
       "      <td>42.267463</td>\n",
       "      <td>47.529076</td>\n",
       "      <td>32.229991</td>\n",
       "      <td>68.258139</td>\n",
       "      <td>57.415641</td>\n",
       "      <td>34.770467</td>\n",
       "      <td>33.378741</td>\n",
       "      <td>67.321897</td>\n",
       "      <td>75.383231</td>\n",
       "      <td>35.938745</td>\n",
       "      <td>59.257508</td>\n",
       "      <td>101.407792</td>\n",
       "      <td>52.536814</td>\n",
       "      <td>46.986842</td>\n",
       "      <td>84.325226</td>\n",
       "      <td>54.736686</td>\n",
       "      <td>41.781448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610007L01Rik</th>\n",
       "      <td>9.941628</td>\n",
       "      <td>10.075592</td>\n",
       "      <td>9.339216</td>\n",
       "      <td>17.118498</td>\n",
       "      <td>12.323579</td>\n",
       "      <td>12.362272</td>\n",
       "      <td>11.723832</td>\n",
       "      <td>13.021746</td>\n",
       "      <td>20.214993</td>\n",
       "      <td>10.277692</td>\n",
       "      <td>13.904486</td>\n",
       "      <td>16.412027</td>\n",
       "      <td>7.248484</td>\n",
       "      <td>10.841421</td>\n",
       "      <td>14.148773</td>\n",
       "      <td>4.960290</td>\n",
       "      <td>11.830840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610007P08Rik</th>\n",
       "      <td>3.723082</td>\n",
       "      <td>7.917463</td>\n",
       "      <td>2.943898</td>\n",
       "      <td>3.989445</td>\n",
       "      <td>6.060253</td>\n",
       "      <td>2.801128</td>\n",
       "      <td>1.278118</td>\n",
       "      <td>4.895650</td>\n",
       "      <td>0.519796</td>\n",
       "      <td>3.377649</td>\n",
       "      <td>4.525034</td>\n",
       "      <td>0.375210</td>\n",
       "      <td>3.264023</td>\n",
       "      <td>8.097296</td>\n",
       "      <td>2.945927</td>\n",
       "      <td>2.475218</td>\n",
       "      <td>6.419092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610007P14Rik</th>\n",
       "      <td>40.772129</td>\n",
       "      <td>49.567579</td>\n",
       "      <td>35.675370</td>\n",
       "      <td>72.237064</td>\n",
       "      <td>43.987477</td>\n",
       "      <td>23.647290</td>\n",
       "      <td>25.025453</td>\n",
       "      <td>11.275608</td>\n",
       "      <td>8.628644</td>\n",
       "      <td>40.701673</td>\n",
       "      <td>9.759840</td>\n",
       "      <td>11.996426</td>\n",
       "      <td>66.573948</td>\n",
       "      <td>63.730330</td>\n",
       "      <td>64.589854</td>\n",
       "      <td>60.966421</td>\n",
       "      <td>47.814948</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     WB3         A1        WB2        GC2       OPC1  \\\n",
       "gene.symbol                                                            \n",
       "0610005C13Rik   0.100000   0.200201   0.100000   0.100000   0.100000   \n",
       "0610007C21Rik  42.267463  47.529076  32.229991  68.258139  57.415641   \n",
       "0610007L01Rik   9.941628  10.075592   9.339216  17.118498  12.323579   \n",
       "0610007P08Rik   3.723082   7.917463   2.943898   3.989445   6.060253   \n",
       "0610007P14Rik  40.772129  49.567579  35.675370  72.237064  43.987477   \n",
       "\n",
       "                      N2         N1      Endo2       MGL1        WB1  \\\n",
       "gene.symbol                                                            \n",
       "0610005C13Rik   0.100000   0.100000   0.100000   0.100000   0.100000   \n",
       "0610007C21Rik  34.770467  33.378741  67.321897  75.383231  35.938745   \n",
       "0610007L01Rik  12.362272  11.723832  13.021746  20.214993  10.277692   \n",
       "0610007P08Rik   2.801128   1.278118   4.895650   0.519796   3.377649   \n",
       "0610007P14Rik  23.647290  25.025453  11.275608   8.628644  40.701673   \n",
       "\n",
       "                   Endo1        MGL2       MOG2        GC1       OPC2  \\\n",
       "gene.symbol                                                             \n",
       "0610005C13Rik   0.100000    0.100000   0.100000   0.100000   0.100000   \n",
       "0610007C21Rik  59.257508  101.407792  52.536814  46.986842  84.325226   \n",
       "0610007L01Rik  13.904486   16.412027   7.248484  10.841421  14.148773   \n",
       "0610007P08Rik   4.525034    0.375210   3.264023   8.097296   2.945927   \n",
       "0610007P14Rik   9.759840   11.996426  66.573948  63.730330  64.589854   \n",
       "\n",
       "                    MOG1         A2  \n",
       "gene.symbol                          \n",
       "0610005C13Rik   0.100000   0.100000  \n",
       "0610007C21Rik  54.736686  41.781448  \n",
       "0610007L01Rik   4.960290  11.830840  \n",
       "0610007P08Rik   2.475218   6.419092  \n",
       "0610007P14Rik  60.966421  47.814948  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(22462, 17)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['WB3', 'A1', 'WB2', 'GC2', 'OPC1', 'N2', 'N1', 'Endo2', 'MGL1', 'WB1',\n",
       "       'Endo1', 'MGL2', 'MOG2', 'GC1', 'OPC2', 'MOG1', 'A2'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "WB3      230264.349011\n",
       "A1       218449.186997\n",
       "WB2      230166.432822\n",
       "GC2      271592.272830\n",
       "OPC1     240692.287928\n",
       "N2       218287.490066\n",
       "N1       205621.040949\n",
       "Endo2    314334.363371\n",
       "MGL1     367933.201803\n",
       "WB1      231959.317925\n",
       "Endo1    294964.022667\n",
       "MGL2     389838.008769\n",
       "MOG2     298078.528164\n",
       "GC1      263362.938329\n",
       "OPC2     248809.707267\n",
       "MOG1     328210.333156\n",
       "A2       216786.270158\n",
       "dtype: float64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_df.sum(axis = 0)# not normalized? or FPKM?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_df.to_pickle(root_directory+'pickle/GSE52564.pickle')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Dataset 2\n",
    "age and sex in data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_2 = root_directory+'GSE60361/GSE60361_C1-3005-Expression.txt'\n",
    "all_data_2 = pd.read_csv(data_2, sep = '\\t', header = 0, index_col = 0) \n",
    "# read count? # how to label cell, paper said based on cluster but they did not offer their clustering result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1772071015_C02</th>\n",
       "      <th>1772071017_G12</th>\n",
       "      <th>1772071017_A05</th>\n",
       "      <th>1772071014_B06</th>\n",
       "      <th>1772067065_H06</th>\n",
       "      <th>1772071017_E02</th>\n",
       "      <th>1772067065_B07</th>\n",
       "      <th>1772067060_B09</th>\n",
       "      <th>1772071014_E04</th>\n",
       "      <th>1772071015_D04</th>\n",
       "      <th>...</th>\n",
       "      <th>1772066110_D12</th>\n",
       "      <th>1772071017_A07</th>\n",
       "      <th>1772063071_G10</th>\n",
       "      <th>1772058148_C03</th>\n",
       "      <th>1772063061_D09</th>\n",
       "      <th>1772067059_B04</th>\n",
       "      <th>1772066097_D04</th>\n",
       "      <th>1772063068_D01</th>\n",
       "      <th>1772066098_A12</th>\n",
       "      <th>1772058148_F03</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Tspan12</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tshz1</th>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Fnbp1l</th>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Adamts15</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cldn12</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 3005 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          1772071015_C02  1772071017_G12  1772071017_A05  1772071014_B06  \\\n",
       "cell_id                                                                    \n",
       "Tspan12                0               0               0               3   \n",
       "Tshz1                  3               1               0               2   \n",
       "Fnbp1l                 3               1               6               4   \n",
       "Adamts15               0               0               0               0   \n",
       "Cldn12                 1               1               1               0   \n",
       "\n",
       "          1772067065_H06  1772071017_E02  1772067065_B07  1772067060_B09  \\\n",
       "cell_id                                                                    \n",
       "Tspan12                0               0               3               0   \n",
       "Tshz1                  2               2               2               1   \n",
       "Fnbp1l                 1               2               1               0   \n",
       "Adamts15               0               0               0               0   \n",
       "Cldn12                 0               0               0               0   \n",
       "\n",
       "          1772071014_E04  1772071015_D04  ...  1772066110_D12  1772071017_A07  \\\n",
       "cell_id                                   ...                                   \n",
       "Tspan12                0               0  ...               0               0   \n",
       "Tshz1                  0               2  ...               0               0   \n",
       "Fnbp1l                 5               2  ...               0               0   \n",
       "Adamts15               0               0  ...               0               0   \n",
       "Cldn12                 2               3  ...               0               0   \n",
       "\n",
       "          1772063071_G10  1772058148_C03  1772063061_D09  1772067059_B04  \\\n",
       "cell_id                                                                    \n",
       "Tspan12                0               0               0               0   \n",
       "Tshz1                  0               0               0               0   \n",
       "Fnbp1l                 0               0               0               0   \n",
       "Adamts15               0               0               0               0   \n",
       "Cldn12                 0               0               0               0   \n",
       "\n",
       "          1772066097_D04  1772063068_D01  1772066098_A12  1772058148_F03  \n",
       "cell_id                                                                   \n",
       "Tspan12                0               0               0               1  \n",
       "Tshz1                  0               0               0               1  \n",
       "Fnbp1l                 0               0               0               0  \n",
       "Adamts15               0               0               0               0  \n",
       "Cldn12                 0               0               0               0  \n",
       "\n",
       "[5 rows x 3005 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_data_2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_data_2.to_pickle(root_directory+'pickle/GSE60631.pickle')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Dataset 3\n",
    "[Original Paper](https://www.pnas.org/content/112/23/7285/tab-figures-data)\n",
    "We used two complementary approaches to classify each single cell to one of the major brain cell types: astrocytes, oligodendrocytes, oligodendrocyte precursor cells (OPCs), neurons, microglia, and vascular cells. First, we used an unbiased approach to sort all 466 individual cells into distinct groups defined by the entirety of their molecular signatures as described in SI Appendix, SI Methods. This approach resulted in the identification of 10 distinct cell groups (unbiased groups) (SI Appendix, Fig. S2). A landscape of all of the cells colored by cell cluster, can be seen in Fig. 1A and SI Appendix, Table S2. Clusters 1–8 consist of adult brain cells, whereas clusters 9 and 10 consist of fetal brain cells.\n",
    "Based upon the presence of typically associated cell-type–specific markers in the top 20 enriched genes for each unbiased group (SI Appendix, Table S3) we were able to easily identify all but two of the clusters as specific groups consisting of OPCs, oligodendrocytes, astrocytes, microglia, neurons, endothelial cells, replicating neuronal progenitors, and quiescent newly born neurons. While these groups contained relatively uniform identities, the enriched genes of two remaining clusters contained genes characteristic of more than one distinct cell type; the interpretation of this finding is discussed below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "466"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_3 = root_directory + 'GSE67835/' \n",
    "filelist = os.listdir(data_3)\n",
    "filelist = [f for f in filelist if 'csv' in f]\n",
    "len(filelist)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_list_3 = []\n",
    "for f in filelist:\n",
    "    # cell id in filename\n",
    "    # although file extension is .csv but it is .tsv the inside\n",
    "    df = pd.read_csv(data_3 + f, sep = '\\t', header = None, names = [f])# a lot of healthy cortex cell\n",
    "    df_list_3.append(df)\n",
    "all_data_3 = pd.concat(df_list_3, axis = 1, ignore_index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GSM1658121_nochipID5.C44.csv</th>\n",
       "      <th>GSM1658106_nochipID3.C72.csv</th>\n",
       "      <th>GSM1658143_nochipID8.C29.csv</th>\n",
       "      <th>GSM1658023_nochipID2.C22.csv</th>\n",
       "      <th>GSM1658155_nochipID8.C49.csv</th>\n",
       "      <th>GSM1657957_1772078236.C70.csv</th>\n",
       "      <th>GSM1658213_nochipID10.C37.csv</th>\n",
       "      <th>GSM1658330_nochipID13.C72.csv</th>\n",
       "      <th>GSM1657895_1772078217.C61.csv</th>\n",
       "      <th>GSM1657873_1772078217.C06.csv</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM1658099_nochipID3.C54.csv</th>\n",
       "      <th>GSM1658229_nochipID11.C02.csv</th>\n",
       "      <th>GSM1658123_nochipID5.C54.csv</th>\n",
       "      <th>GSM1658118_nochipID5.C18.csv</th>\n",
       "      <th>GSM1657871_1772078217.C03.csv</th>\n",
       "      <th>GSM1658358_nochipID4.C63.csv</th>\n",
       "      <th>GSM1658094_nochipID3.C37.csv</th>\n",
       "      <th>GSM1658012_nochipID2.C10.csv</th>\n",
       "      <th>GSM1658328_nochipID13.C67.csv</th>\n",
       "      <th>GSM1658240_nochipID11.C22.csv</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1/2-SBSRNA4</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>72</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2LD1</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 466 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              GSM1658121_nochipID5.C44.csv  GSM1658106_nochipID3.C72.csv  \\\n",
       "1/2-SBSRNA4                              0                             0   \n",
       "A1BG                                     0                             0   \n",
       "A1BG-AS1                                 0                             0   \n",
       "A1CF                                     0                             0   \n",
       "A2LD1                                    0                             0   \n",
       "\n",
       "              GSM1658143_nochipID8.C29.csv  GSM1658023_nochipID2.C22.csv  \\\n",
       "1/2-SBSRNA4                              0                             0   \n",
       "A1BG                                     0                             0   \n",
       "A1BG-AS1                                 0                             0   \n",
       "A1CF                                     0                             0   \n",
       "A2LD1                                    0                             7   \n",
       "\n",
       "              GSM1658155_nochipID8.C49.csv  GSM1657957_1772078236.C70.csv  \\\n",
       "1/2-SBSRNA4                              0                              0   \n",
       "A1BG                                     7                              0   \n",
       "A1BG-AS1                                 0                              0   \n",
       "A1CF                                     0                              0   \n",
       "A2LD1                                    0                              0   \n",
       "\n",
       "              GSM1658213_nochipID10.C37.csv  GSM1658330_nochipID13.C72.csv  \\\n",
       "1/2-SBSRNA4                               0                              0   \n",
       "A1BG                                      0                              0   \n",
       "A1BG-AS1                                  0                              0   \n",
       "A1CF                                      0                              0   \n",
       "A2LD1                                     0                              0   \n",
       "\n",
       "              GSM1657895_1772078217.C61.csv  GSM1657873_1772078217.C06.csv  \\\n",
       "1/2-SBSRNA4                               2                              0   \n",
       "A1BG                                      0                              0   \n",
       "A1BG-AS1                                  0                              0   \n",
       "A1CF                                      0                              0   \n",
       "A2LD1                                     0                              0   \n",
       "\n",
       "              ...  GSM1658099_nochipID3.C54.csv  \\\n",
       "1/2-SBSRNA4   ...                             0   \n",
       "A1BG          ...                             0   \n",
       "A1BG-AS1      ...                             0   \n",
       "A1CF          ...                             0   \n",
       "A2LD1         ...                             0   \n",
       "\n",
       "              GSM1658229_nochipID11.C02.csv  GSM1658123_nochipID5.C54.csv  \\\n",
       "1/2-SBSRNA4                              72                             0   \n",
       "A1BG                                      0                            15   \n",
       "A1BG-AS1                                  0                             0   \n",
       "A1CF                                      0                             0   \n",
       "A2LD1                                     0                             0   \n",
       "\n",
       "              GSM1658118_nochipID5.C18.csv  GSM1657871_1772078217.C03.csv  \\\n",
       "1/2-SBSRNA4                              0                              0   \n",
       "A1BG                                     0                              0   \n",
       "A1BG-AS1                                 0                              0   \n",
       "A1CF                                     0                              0   \n",
       "A2LD1                                    0                              0   \n",
       "\n",
       "              GSM1658358_nochipID4.C63.csv  GSM1658094_nochipID3.C37.csv  \\\n",
       "1/2-SBSRNA4                              0                             0   \n",
       "A1BG                                     0                             0   \n",
       "A1BG-AS1                                 0                             0   \n",
       "A1CF                                     0                             0   \n",
       "A2LD1                                    0                             0   \n",
       "\n",
       "              GSM1658012_nochipID2.C10.csv  GSM1658328_nochipID13.C67.csv  \\\n",
       "1/2-SBSRNA4                              0                              0   \n",
       "A1BG                                     0                              0   \n",
       "A1BG-AS1                                 0                              0   \n",
       "A1CF                                     0                              0   \n",
       "A2LD1                                    0                              0   \n",
       "\n",
       "              GSM1658240_nochipID11.C22.csv  \n",
       "1/2-SBSRNA4                               0  \n",
       "A1BG                                      0  \n",
       "A1BG-AS1                                  0  \n",
       "A1CF                                      0  \n",
       "A2LD1                                     0  \n",
       "\n",
       "[5 rows x 466 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_data_3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GSM1658121_nochipID5.C44.csv     1056873\n",
       "GSM1658106_nochipID3.C72.csv     2138110\n",
       "GSM1658143_nochipID8.C29.csv     2490339\n",
       "GSM1658023_nochipID2.C22.csv     3101242\n",
       "GSM1658155_nochipID8.C49.csv     1972138\n",
       "                                  ...   \n",
       "GSM1658358_nochipID4.C63.csv     1786056\n",
       "GSM1658094_nochipID3.C37.csv     2098165\n",
       "GSM1658012_nochipID2.C10.csv     1642874\n",
       "GSM1658328_nochipID13.C67.csv    1735585\n",
       "GSM1658240_nochipID11.C22.csv    1645206\n",
       "Length: 466, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_data_3.sum() # count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['1/2-SBSRNA4 ', 'A1BG ', 'A1BG-AS1 ', 'A1CF ', 'A2LD1 ', 'A2M ',\n",
       "       'A2ML1 ', 'A2MP1 ', 'A4GALT ', 'A4GNT ',\n",
       "       ...\n",
       "       'ZXDC ', 'ZYG11A ', 'ZYG11B ', 'ZYX ', 'ZZEF1 ', 'ZZZ3 ', 'tAKR ',\n",
       "       'no_feature ', 'ambiguous ', 'alignment_not_unique '],\n",
       "      dtype='object', length=22088)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_data_3.index # three rows that are not genes 'no_feature ', 'ambiguous ', 'alignment_not_unique '"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_data_3.to_pickle(root_directory+'pickle/GSE67835.pickle')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data 4\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_4 = root_directory+'GSE71585/GSE71585_RefSeq_counts.csv'\n",
    "all_data_4 = pd.read_csv(data_4, header = 0, index_col = 0)\n",
    "data_4_cluster = root_directory+'GSE71585/GSE71585_Clustering_Results.csv' "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Calb2_tdTpositive_cell_1</th>\n",
       "      <th>Calb2_tdTpositive_cell_2</th>\n",
       "      <th>Calb2_tdTpositive_cell_3</th>\n",
       "      <th>Calb2_tdTpositive_cell_4</th>\n",
       "      <th>Calb2_tdTpositive_cell_5</th>\n",
       "      <th>Calb2_tdTpositive_cell_6</th>\n",
       "      <th>Calb2_tdTpositive_cell_7</th>\n",
       "      <th>Calb2_tdTpositive_cell_8</th>\n",
       "      <th>Calb2_tdTpositive_cell_9</th>\n",
       "      <th>Calb2_tdTpositive_cell_10</th>\n",
       "      <th>...</th>\n",
       "      <th>CAV_VISp_Contra_tdTpos_cell_5</th>\n",
       "      <th>Rbp4_CTX_10pg_1</th>\n",
       "      <th>Rbp4_CTX_10pg_2</th>\n",
       "      <th>Rbp4_CTX_10pg_3</th>\n",
       "      <th>Rbp4_CTX_10pg_4</th>\n",
       "      <th>Rbp4_CTX_10pg_5</th>\n",
       "      <th>Rbp4_CTX_10pg_6</th>\n",
       "      <th>Rbp4_CTX_250ng_1</th>\n",
       "      <th>Rbp4_CTX_250ng_2</th>\n",
       "      <th>Trib2_CTX_250ng_1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0610005C13Rik</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>8.00</td>\n",
       "      <td>9.00</td>\n",
       "      <td>5.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610007C21Rik</th>\n",
       "      <td>992.00</td>\n",
       "      <td>2287.02</td>\n",
       "      <td>491.78</td>\n",
       "      <td>1932.0</td>\n",
       "      <td>1425.0</td>\n",
       "      <td>130.03</td>\n",
       "      <td>2110.02</td>\n",
       "      <td>955.0</td>\n",
       "      <td>326.0</td>\n",
       "      <td>933.0</td>\n",
       "      <td>...</td>\n",
       "      <td>352.06</td>\n",
       "      <td>401.0</td>\n",
       "      <td>1750.99</td>\n",
       "      <td>1463.0</td>\n",
       "      <td>3439.0</td>\n",
       "      <td>502.95</td>\n",
       "      <td>1486.36</td>\n",
       "      <td>956.40</td>\n",
       "      <td>1054.94</td>\n",
       "      <td>944.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610007L01Rik</th>\n",
       "      <td>2.57</td>\n",
       "      <td>177.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.00</td>\n",
       "      <td>3040.99</td>\n",
       "      <td>101.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1042.0</td>\n",
       "      <td>...</td>\n",
       "      <td>367.00</td>\n",
       "      <td>242.0</td>\n",
       "      <td>1.00</td>\n",
       "      <td>69.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>53.00</td>\n",
       "      <td>40.15</td>\n",
       "      <td>1251.00</td>\n",
       "      <td>1356.20</td>\n",
       "      <td>1147.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610007N19Rik</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>704.01</td>\n",
       "      <td>0.00</td>\n",
       "      <td>27.00</td>\n",
       "      <td>37.00</td>\n",
       "      <td>35.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610007P08Rik</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>17.03</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>3.00</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>574.35</td>\n",
       "      <td>570.86</td>\n",
       "      <td>590.84</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 1809 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Calb2_tdTpositive_cell_1  Calb2_tdTpositive_cell_2  \\\n",
       "gene                                                                \n",
       "0610005C13Rik                      0.00                      0.00   \n",
       "0610007C21Rik                    992.00                   2287.02   \n",
       "0610007L01Rik                      2.57                    177.00   \n",
       "0610007N19Rik                      0.00                      0.00   \n",
       "0610007P08Rik                      0.00                      0.00   \n",
       "\n",
       "               Calb2_tdTpositive_cell_3  Calb2_tdTpositive_cell_4  \\\n",
       "gene                                                                \n",
       "0610005C13Rik                      0.00                       0.0   \n",
       "0610007C21Rik                    491.78                    1932.0   \n",
       "0610007L01Rik                      0.00                       1.0   \n",
       "0610007N19Rik                      0.00                       0.0   \n",
       "0610007P08Rik                      0.00                       0.0   \n",
       "\n",
       "               Calb2_tdTpositive_cell_5  Calb2_tdTpositive_cell_6  \\\n",
       "gene                                                                \n",
       "0610005C13Rik                       0.0                      0.00   \n",
       "0610007C21Rik                    1425.0                    130.03   \n",
       "0610007L01Rik                       2.0                      3.00   \n",
       "0610007N19Rik                       0.0                      0.00   \n",
       "0610007P08Rik                       0.0                      0.00   \n",
       "\n",
       "               Calb2_tdTpositive_cell_7  Calb2_tdTpositive_cell_8  \\\n",
       "gene                                                                \n",
       "0610005C13Rik                      0.00                       0.0   \n",
       "0610007C21Rik                   2110.02                     955.0   \n",
       "0610007L01Rik                   3040.99                     101.0   \n",
       "0610007N19Rik                      0.00                       0.0   \n",
       "0610007P08Rik                     17.03                       0.0   \n",
       "\n",
       "               Calb2_tdTpositive_cell_9  Calb2_tdTpositive_cell_10  ...  \\\n",
       "gene                                                                ...   \n",
       "0610005C13Rik                       0.0                        0.0  ...   \n",
       "0610007C21Rik                     326.0                      933.0  ...   \n",
       "0610007L01Rik                       0.0                     1042.0  ...   \n",
       "0610007N19Rik                       0.0                        0.0  ...   \n",
       "0610007P08Rik                       0.0                        0.0  ...   \n",
       "\n",
       "               CAV_VISp_Contra_tdTpos_cell_5  Rbp4_CTX_10pg_1  \\\n",
       "gene                                                            \n",
       "0610005C13Rik                           0.00              0.0   \n",
       "0610007C21Rik                         352.06            401.0   \n",
       "0610007L01Rik                         367.00            242.0   \n",
       "0610007N19Rik                           0.00              0.0   \n",
       "0610007P08Rik                           3.00             11.0   \n",
       "\n",
       "               Rbp4_CTX_10pg_2  Rbp4_CTX_10pg_3  Rbp4_CTX_10pg_4  \\\n",
       "gene                                                               \n",
       "0610005C13Rik             0.00              0.0              0.0   \n",
       "0610007C21Rik          1750.99           1463.0           3439.0   \n",
       "0610007L01Rik             1.00             69.0              0.0   \n",
       "0610007N19Rik             0.00              0.0              0.0   \n",
       "0610007P08Rik             0.00              0.0              0.0   \n",
       "\n",
       "               Rbp4_CTX_10pg_5  Rbp4_CTX_10pg_6  Rbp4_CTX_250ng_1  \\\n",
       "gene                                                                \n",
       "0610005C13Rik             0.00             0.00              8.00   \n",
       "0610007C21Rik           502.95          1486.36            956.40   \n",
       "0610007L01Rik            53.00            40.15           1251.00   \n",
       "0610007N19Rik           704.01             0.00             27.00   \n",
       "0610007P08Rik             0.00             0.00            574.35   \n",
       "\n",
       "               Rbp4_CTX_250ng_2  Trib2_CTX_250ng_1  \n",
       "gene                                                \n",
       "0610005C13Rik              9.00               5.00  \n",
       "0610007C21Rik           1054.94             944.03  \n",
       "0610007L01Rik           1356.20            1147.00  \n",
       "0610007N19Rik             37.00              35.00  \n",
       "0610007P08Rik            570.86             590.84  \n",
       "\n",
       "[5 rows x 1809 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_data_4.head() # why counts have digital?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mouse_line</th>\n",
       "      <th>cre_driver_1</th>\n",
       "      <th>cre_driver_2</th>\n",
       "      <th>cre_reporter</th>\n",
       "      <th>dissection</th>\n",
       "      <th>tdTomato</th>\n",
       "      <th>pass_qc_checks</th>\n",
       "      <th>broad_type</th>\n",
       "      <th>core_intermediate</th>\n",
       "      <th>primary_type</th>\n",
       "      <th>secondary_type</th>\n",
       "      <th>aibs_vignette_id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_title</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Calb2_tdTpositive_cell_46</th>\n",
       "      <td>Calb2</td>\n",
       "      <td>Calb2-IRES-Cre</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RCL-tdT (Ai14)</td>\n",
       "      <td>upper</td>\n",
       "      <td>positive</td>\n",
       "      <td>Y</td>\n",
       "      <td>Astrocyte</td>\n",
       "      <td>core</td>\n",
       "      <td>Astro Gja1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1631_VU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Calb2_tdTpositive_cell_48</th>\n",
       "      <td>Calb2</td>\n",
       "      <td>Calb2-IRES-Cre</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RCL-tdT (Ai14)</td>\n",
       "      <td>upper</td>\n",
       "      <td>positive</td>\n",
       "      <td>Y</td>\n",
       "      <td>Astrocyte</td>\n",
       "      <td>core</td>\n",
       "      <td>Astro Gja1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1633_VU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Calb2_tdTpositive_cell_50</th>\n",
       "      <td>Calb2</td>\n",
       "      <td>Calb2-IRES-Cre</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RCL-tdT (Ai14)</td>\n",
       "      <td>lower</td>\n",
       "      <td>positive</td>\n",
       "      <td>Y</td>\n",
       "      <td>Astrocyte</td>\n",
       "      <td>core</td>\n",
       "      <td>Astro Gja1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1635_VL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Calb2_tdTpositive_cell_53</th>\n",
       "      <td>Calb2</td>\n",
       "      <td>Calb2-IRES-Cre</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RCL-tdT (Ai14)</td>\n",
       "      <td>lower</td>\n",
       "      <td>positive</td>\n",
       "      <td>Y</td>\n",
       "      <td>Astrocyte</td>\n",
       "      <td>core</td>\n",
       "      <td>Astro Gja1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1638_VL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Calb2_tdTpositive_cell_58</th>\n",
       "      <td>Calb2</td>\n",
       "      <td>Calb2-IRES-Cre</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RCL-tdT (Ai14)</td>\n",
       "      <td>lower</td>\n",
       "      <td>positive</td>\n",
       "      <td>Y</td>\n",
       "      <td>Astrocyte</td>\n",
       "      <td>core</td>\n",
       "      <td>Astro Gja1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A1643_VL</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          mouse_line    cre_driver_1 cre_driver_2  \\\n",
       "sample_title                                                        \n",
       "Calb2_tdTpositive_cell_46      Calb2  Calb2-IRES-Cre          NaN   \n",
       "Calb2_tdTpositive_cell_48      Calb2  Calb2-IRES-Cre          NaN   \n",
       "Calb2_tdTpositive_cell_50      Calb2  Calb2-IRES-Cre          NaN   \n",
       "Calb2_tdTpositive_cell_53      Calb2  Calb2-IRES-Cre          NaN   \n",
       "Calb2_tdTpositive_cell_58      Calb2  Calb2-IRES-Cre          NaN   \n",
       "\n",
       "                             cre_reporter dissection  tdTomato pass_qc_checks  \\\n",
       "sample_title                                                                    \n",
       "Calb2_tdTpositive_cell_46  RCL-tdT (Ai14)      upper  positive              Y   \n",
       "Calb2_tdTpositive_cell_48  RCL-tdT (Ai14)      upper  positive              Y   \n",
       "Calb2_tdTpositive_cell_50  RCL-tdT (Ai14)      lower  positive              Y   \n",
       "Calb2_tdTpositive_cell_53  RCL-tdT (Ai14)      lower  positive              Y   \n",
       "Calb2_tdTpositive_cell_58  RCL-tdT (Ai14)      lower  positive              Y   \n",
       "\n",
       "                          broad_type core_intermediate primary_type  \\\n",
       "sample_title                                                          \n",
       "Calb2_tdTpositive_cell_46  Astrocyte              core   Astro Gja1   \n",
       "Calb2_tdTpositive_cell_48  Astrocyte              core   Astro Gja1   \n",
       "Calb2_tdTpositive_cell_50  Astrocyte              core   Astro Gja1   \n",
       "Calb2_tdTpositive_cell_53  Astrocyte              core   Astro Gja1   \n",
       "Calb2_tdTpositive_cell_58  Astrocyte              core   Astro Gja1   \n",
       "\n",
       "                          secondary_type aibs_vignette_id  \n",
       "sample_title                                               \n",
       "Calb2_tdTpositive_cell_46            NaN         A1631_VU  \n",
       "Calb2_tdTpositive_cell_48            NaN         A1633_VU  \n",
       "Calb2_tdTpositive_cell_50            NaN         A1635_VL  \n",
       "Calb2_tdTpositive_cell_53            NaN         A1638_VL  \n",
       "Calb2_tdTpositive_cell_58            NaN         A1643_VL  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_data_4_cluster = pd.read_csv(data_4_cluster, index_col = 0)\n",
    "all_data_4_cluster.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Calb2_tdTpositive_cell_1    14030202.14\n",
       "Calb2_tdTpositive_cell_2     4803366.44\n",
       "Calb2_tdTpositive_cell_3     3073418.37\n",
       "Calb2_tdTpositive_cell_4    12192899.59\n",
       "Calb2_tdTpositive_cell_5    13477232.54\n",
       "                               ...     \n",
       "Rbp4_CTX_10pg_5             13166764.03\n",
       "Rbp4_CTX_10pg_6             10340431.59\n",
       "Rbp4_CTX_250ng_1            25486493.33\n",
       "Rbp4_CTX_250ng_2            26684731.91\n",
       "Trib2_CTX_250ng_1           24775226.25\n",
       "Length: 1809, dtype: float64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_data_4.sum() # ???? #?????"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_data_4.to_pickle(root_directory+'pickle/GSE71585.pickle')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data 5\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_5 = root_directory+'GSE73721/GSE73721_Human_and_mouse_table.csv'\n",
    "all_data_5 = pd.read_csv(data_5, header = 0, index_col = 0) # normalized count?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>59yo tumor periphery astro</th>\n",
       "      <th>59yo tumor core astro</th>\n",
       "      <th>64yo tumor core astro</th>\n",
       "      <th>65yo tumor core astro</th>\n",
       "      <th>21yo Hippocampus astro</th>\n",
       "      <th>22yo Hippocampus astro</th>\n",
       "      <th>53yo A Hippocampus astro</th>\n",
       "      <th>53yo B Hippocampus astro</th>\n",
       "      <th>Fetal ctx 1 astro</th>\n",
       "      <th>Fetal ctx 2 astro</th>\n",
       "      <th>...</th>\n",
       "      <th>47 yo ctx endo</th>\n",
       "      <th>13yo ctx endo</th>\n",
       "      <th>45yo whole cortex</th>\n",
       "      <th>63yo whole cortex</th>\n",
       "      <th>25yo whole cortex</th>\n",
       "      <th>53yo whole cortex</th>\n",
       "      <th>Mouse 1 month astro</th>\n",
       "      <th>Mouse 4 month astro</th>\n",
       "      <th>Mouse 7 month astro</th>\n",
       "      <th>Mouse 9 month astro</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Aamp</th>\n",
       "      <td>0.1</td>\n",
       "      <td>1.1</td>\n",
       "      <td>0.7</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.4</td>\n",
       "      <td>3.4</td>\n",
       "      <td>2.1</td>\n",
       "      <td>0.7</td>\n",
       "      <td>2.8</td>\n",
       "      <td>0.1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.2</td>\n",
       "      <td>0.5</td>\n",
       "      <td>1.2</td>\n",
       "      <td>1.4</td>\n",
       "      <td>5.3</td>\n",
       "      <td>5.5</td>\n",
       "      <td>9.6</td>\n",
       "      <td>4.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abat</th>\n",
       "      <td>11.0</td>\n",
       "      <td>17.2</td>\n",
       "      <td>12.3</td>\n",
       "      <td>37.0</td>\n",
       "      <td>16.1</td>\n",
       "      <td>12.3</td>\n",
       "      <td>10.2</td>\n",
       "      <td>9.4</td>\n",
       "      <td>8.5</td>\n",
       "      <td>17.4</td>\n",
       "      <td>...</td>\n",
       "      <td>1.2</td>\n",
       "      <td>2.5</td>\n",
       "      <td>11.6</td>\n",
       "      <td>6.7</td>\n",
       "      <td>14.5</td>\n",
       "      <td>9.7</td>\n",
       "      <td>54.9</td>\n",
       "      <td>50.9</td>\n",
       "      <td>70.3</td>\n",
       "      <td>56.8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abca1</th>\n",
       "      <td>2.1</td>\n",
       "      <td>35.2</td>\n",
       "      <td>11.2</td>\n",
       "      <td>27.5</td>\n",
       "      <td>8.8</td>\n",
       "      <td>10.1</td>\n",
       "      <td>6.5</td>\n",
       "      <td>4.8</td>\n",
       "      <td>0.9</td>\n",
       "      <td>1.5</td>\n",
       "      <td>...</td>\n",
       "      <td>1.3</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.1</td>\n",
       "      <td>1.6</td>\n",
       "      <td>1.7</td>\n",
       "      <td>1.8</td>\n",
       "      <td>9.6</td>\n",
       "      <td>6.4</td>\n",
       "      <td>13.6</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abca2</th>\n",
       "      <td>0.5</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.3</td>\n",
       "      <td>7.7</td>\n",
       "      <td>0.2</td>\n",
       "      <td>8.2</td>\n",
       "      <td>8.5</td>\n",
       "      <td>1.7</td>\n",
       "      <td>0.4</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.4</td>\n",
       "      <td>0.9</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.2</td>\n",
       "      <td>3.2</td>\n",
       "      <td>5.9</td>\n",
       "      <td>6.4</td>\n",
       "      <td>5.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Abcc5</th>\n",
       "      <td>2.1</td>\n",
       "      <td>3.2</td>\n",
       "      <td>2.3</td>\n",
       "      <td>5.7</td>\n",
       "      <td>3.7</td>\n",
       "      <td>5.7</td>\n",
       "      <td>4.1</td>\n",
       "      <td>5.1</td>\n",
       "      <td>2.4</td>\n",
       "      <td>7.6</td>\n",
       "      <td>...</td>\n",
       "      <td>0.6</td>\n",
       "      <td>1.1</td>\n",
       "      <td>3.8</td>\n",
       "      <td>1.7</td>\n",
       "      <td>5.9</td>\n",
       "      <td>1.5</td>\n",
       "      <td>5.3</td>\n",
       "      <td>2.3</td>\n",
       "      <td>10.3</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 45 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       59yo tumor periphery astro  59yo tumor core astro  \\\n",
       "Gene                                                       \n",
       "Aamp                          0.1                    1.1   \n",
       "Abat                         11.0                   17.2   \n",
       "Abca1                         2.1                   35.2   \n",
       "Abca2                         0.5                    0.6   \n",
       "Abcc5                         2.1                    3.2   \n",
       "\n",
       "       64yo tumor core astro  65yo tumor core astro  21yo Hippocampus astro  \\\n",
       "Gene                                                                          \n",
       "Aamp                     0.7                    1.5                     1.4   \n",
       "Abat                    12.3                   37.0                    16.1   \n",
       "Abca1                   11.2                   27.5                     8.8   \n",
       "Abca2                    0.6                    0.3                     7.7   \n",
       "Abcc5                    2.3                    5.7                     3.7   \n",
       "\n",
       "       22yo Hippocampus astro  53yo A Hippocampus astro  \\\n",
       "Gene                                                      \n",
       "Aamp                      3.4                       2.1   \n",
       "Abat                     12.3                      10.2   \n",
       "Abca1                    10.1                       6.5   \n",
       "Abca2                     0.2                       8.2   \n",
       "Abcc5                     5.7                       4.1   \n",
       "\n",
       "       53yo B Hippocampus astro  Fetal ctx 1 astro  Fetal ctx 2 astro  ...  \\\n",
       "Gene                                                                   ...   \n",
       "Aamp                        0.7                2.8                0.1  ...   \n",
       "Abat                        9.4                8.5               17.4  ...   \n",
       "Abca1                       4.8                0.9                1.5  ...   \n",
       "Abca2                       8.5                1.7                0.4  ...   \n",
       "Abcc5                       5.1                2.4                7.6  ...   \n",
       "\n",
       "       47 yo ctx endo  13yo ctx endo  45yo whole cortex  63yo whole cortex  \\\n",
       "Gene                                                                         \n",
       "Aamp              0.2            1.0                1.2                0.5   \n",
       "Abat              1.2            2.5               11.6                6.7   \n",
       "Abca1             1.3            1.5                1.1                1.6   \n",
       "Abca2             0.1            0.4                0.9                0.1   \n",
       "Abcc5             0.6            1.1                3.8                1.7   \n",
       "\n",
       "       25yo whole cortex  53yo whole cortex  Mouse 1 month astro  \\\n",
       "Gene                                                               \n",
       "Aamp                 1.2                1.4                  5.3   \n",
       "Abat                14.5                9.7                 54.9   \n",
       "Abca1                1.7                1.8                  9.6   \n",
       "Abca2                0.5                0.2                  3.2   \n",
       "Abcc5                5.9                1.5                  5.3   \n",
       "\n",
       "       Mouse 4 month astro  Mouse 7 month astro  Mouse 9 month astro  \n",
       "Gene                                                                  \n",
       "Aamp                   5.5                  9.6                  4.4  \n",
       "Abat                  50.9                 70.3                 56.8  \n",
       "Abca1                  6.4                 13.6                  9.0  \n",
       "Abca2                  5.9                  6.4                  5.4  \n",
       "Abcc5                  2.3                 10.3                  3.0  \n",
       "\n",
       "[5 rows x 45 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_data_5.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "59yo tumor periphery astro     50527.1\n",
       "59yo tumor core astro         103290.4\n",
       "64yo tumor core astro          73933.8\n",
       "65yo tumor core astro          74880.2\n",
       "21yo Hippocampus astro         94227.1\n",
       "22yo Hippocampus astro         85454.0\n",
       "53yo A Hippocampus astro       98480.4\n",
       "53yo B Hippocampus astro       79043.4\n",
       "Fetal ctx 1 astro              61928.6\n",
       "Fetal ctx 2 astro              84277.1\n",
       "Fetal ctx 3 astro              74515.0\n",
       "Fetal ctx 4 astro              98840.4\n",
       "Fetal ctx 5 astro              58122.7\n",
       "Fetal ctx 6 astro              73975.4\n",
       "8yo ctx astro                  89389.9\n",
       "13yo ctx astro                 85544.1\n",
       "16yo ctx astro                 86464.6\n",
       "21yo ctx astro                 81246.4\n",
       "22yo ctx astro                 99001.4\n",
       "35yo ctx astro                 96832.6\n",
       "47yo ctx astro                 71864.7\n",
       "51yo ctx astro                 90287.2\n",
       "53yo ctx astro                 74999.9\n",
       "60yo ctx astro                 65163.4\n",
       "63yo ctx 1 astro               80508.2\n",
       "63yo ctx 2 astro               72712.5\n",
       "25yo ctx neuron                95245.3\n",
       "22yo ctx oligo                 77919.3\n",
       "63yo ctx A oligo               78084.8\n",
       "63yo ctx B oligo              106484.5\n",
       "47yo ctx oligo                 81116.6\n",
       "63yo ctx 3 oligo               67548.5\n",
       "45yo ctx myeloid               69145.7\n",
       "51yo ctx myeloid              103811.1\n",
       "63 yo ctx myeloid             122737.0\n",
       "47 yo ctx endo                 48808.5\n",
       "13yo ctx endo                  40884.8\n",
       "45yo whole cortex              37032.9\n",
       "63yo whole cortex              34140.5\n",
       "25yo whole cortex              39067.1\n",
       "53yo whole cortex              36201.4\n",
       "Mouse 1 month astro           111369.6\n",
       "Mouse 4 month astro            97095.4\n",
       "Mouse 7 month astro           124090.9\n",
       "Mouse 9 month astro           110934.3\n",
       "dtype: float64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_data_5.sum() #FPKM? From epilepsy patient"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_data_5.to_pickle(root_directory+'pickle/GSE73721.pickle')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Result: Data characteristic\n",
    "Dataset are processed by different pipeline.\n",
    "Some \"cell type\" labels are data driven (not based on immune markers/FACs)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
