{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "def new_slurm_script(script_name, cpus_per_task='1', mem_per_cpu='5G', array_t='1-9000', array_tc='250'):\n",
    "\n",
    "\n",
    "\n",
    "    \n",
    "    with open(script_name, 'w') as f:\n",
    "        f.write('''#! /bin/bash\n",
    "        \n",
    "#SBATCH --job-name={0}\n",
    "#SBATCH --output={0}.out\n",
    "#SBATCH --error={0}.error\n",
    "#SBATCH --cpus-per-task={1}\n",
    "#SBATCH --mem-per-cpu={2}\n",
    "#SBATCH --array={3}%{4}\n",
    "'''.format(script_name, cpus_per_task, mem_per_cpu, array_t, array_tc))\n",
    "\n",
    "\n",
    "def map_slurm_task_id(script_name, file_list, var_name):\n",
    "\n",
    "    # generate file list with ' '\n",
    "    file_list_str = ' '.join(file_list)\n",
    "\n",
    "\n",
    "    with open(script_name, 'a') as f:\n",
    "        f.write('''\n",
    "{0}s=({1})\n",
    "{0}=${{{0}s[$SLURM_ARRAY_TASK_ID -1]}}\n",
    "'''.format(var_name, file_list_str))\n",
    "\n",
    "def unzip_all(script_name, var_name):\n",
    "    with open(script_name, 'a') as f:\n",
    "        f.write('''\n",
    "gunzip ${0}'''.format(var_name))\n",
    "def star_cmd(script_name, genome, fastq_var, output_path):\n",
    "    with open(script_name, 'a') as f:\n",
    "        f.write('''\n",
    "STAR --genomeDir {0} \\\n",
    "--runThreadN 8 \\\n",
    "--readFilesIn {1} \\\n",
    "--outFileNamePrefix {2}/$ID \\\n",
    "--outSAMtype BAM SortedByCoordinate \\\n",
    "--outSAMunmapped Within \\\n",
    "--outSAMattributes Standard \\\n",
    "--outTmpDir ./_STARtmp$ID \\\n",
    "\n",
    "\n",
    "'''.format(genome, fastq_var, output_path))\n",
    "def feature_count(gtf, outfile, infile_list, script_name):\n",
    "    with open(script_name, 'a') as f:\n",
    "        f.write('''\n",
    "featureCounts -t exon -g gene_id -a {0} -o {1} {2}\n",
    "'''.format(gtf, outfile, ' '.join(infile_list)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "root_dir = '/cellar/users/hsher/Data/ideker/sc/'\n",
    "human_genome_index = '/nrnb/users/hsher/star_hg38'\n",
    "mouse_genome_index = '/nrnb/users/hsher/star_mous'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "rnaseq = pd.read_excel('RNA-seq.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>GEO accession</th>\n",
       "      <th>species</th>\n",
       "      <th>SRA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Darmanis</td>\n",
       "      <td>GSE67835</td>\n",
       "      <td>human</td>\n",
       "      <td>SRP057196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Zhang</td>\n",
       "      <td>GSE73721</td>\n",
       "      <td>human</td>\n",
       "      <td>SRP064454</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Zhang</td>\n",
       "      <td>GSE52564</td>\n",
       "      <td>mouse</td>\n",
       "      <td>SRP033200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Zeisel</td>\n",
       "      <td>GSE60361</td>\n",
       "      <td>mouse</td>\n",
       "      <td>SRP045452</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Tasic</td>\n",
       "      <td>GSE71585</td>\n",
       "      <td>mouse</td>\n",
       "      <td>SRP061902</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Unnamed: 0 GEO accession species        SRA\n",
       "0   Darmanis      GSE67835   human  SRP057196\n",
       "1      Zhang      GSE73721   human  SRP064454\n",
       "2      Zhang      GSE52564   mouse  SRP033200\n",
       "3     Zeisel      GSE60361   mouse  SRP045452\n",
       "4      Tasic      GSE71585   mouse  SRP061902"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rnaseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "number = 0\n",
    "\n",
    "accession = rnaseq.loc[number, 'GEO accession']\n",
    "srp_id = rnaseq.loc[number, 'SRA']\n",
    "sps = rnaseq.loc[number, 'species']\n",
    "\n",
    "fastq_path = [os.path.join(os.path.join(root_dir, srp_id), file) for file in os.listdir(os.path.join(root_dir, srp_id))]\n",
    "prefix = list(set([srr.split('_')[0] for srr in fastq_path]))\n",
    "srr_id = [p.split('/')[-1] for p in prefix]\n",
    "\n",
    "sc_name = accession+'_star.run'\n",
    "new_slurm_script(sc_name, '8', '8G', '1-{}'.format(len(srr_id)), '30')\n",
    "map_slurm_task_id(sc_name, [p+ '_1.fastq' for p in prefix], 'FASTQ_ONE')\n",
    "map_slurm_task_id(sc_name, [p+ '_2.fastq' for p in prefix], 'FASTQ_TWO')\n",
    "map_slurm_task_id(sc_name, srr_id, 'ID')\n",
    "\n",
    "if sps == 'human':\n",
    "    star_cmd(sc_name, human_genome_index, '$FASTQ_ONE $FASTQ_TWO', '/nrnb/users/hsher/{}'.format(accession))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "number = 1\n",
    "\n",
    "accession = rnaseq.loc[number, 'GEO accession']\n",
    "srp_id = rnaseq.loc[number, 'SRA']\n",
    "sps = rnaseq.loc[number, 'species']\n",
    "\n",
    "fastq_path = [os.path.join(os.path.join(root_dir, srp_id), file) for file in os.listdir(os.path.join(root_dir, srp_id))]\n",
    "prefix = list(set([srr.split('_')[0] for srr in fastq_path]))\n",
    "srr_id = [p.split('/')[-1] for p in prefix]\n",
    "\n",
    "sc_name = accession+'_star.run'\n",
    "new_slurm_script(sc_name, '8', '8G', '1-{}'.format(len(srr_id)), '30')\n",
    "map_slurm_task_id(sc_name, [p+ '_1.fastq' for p in prefix], 'FASTQ_ONE')\n",
    "map_slurm_task_id(sc_name, [p+ '_2.fastq' for p in prefix], 'FASTQ_TWO')\n",
    "map_slurm_task_id(sc_name, srr_id, 'ID')\n",
    "\n",
    "if sps == 'human':\n",
    "    star_cmd(sc_name, human_genome_index, '$FASTQ_ONE $FASTQ_TWO', '/nrnb/users/hsher/{}'.format(accession))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "number = 4\n",
    "\n",
    "accession = rnaseq.loc[number, 'GEO accession']\n",
    "srp_id = rnaseq.loc[number, 'SRA']\n",
    "sps = rnaseq.loc[number, 'species']\n",
    "\n",
    "fastq_path = [os.path.join(os.path.join(root_dir, srp_id), file) for file in os.listdir(os.path.join(root_dir, srp_id))]\n",
    "prefix = list(set([srr.split('_')[0] for srr in fastq_path]))\n",
    "srr_id = [p.split('/')[-1].replace('.fastq', '') for p in prefix]\n",
    "\n",
    "sc_name = accession+'_star.run'\n",
    "new_slurm_script(sc_name, '8', '8G', '1-{}'.format(len(srr_id)), '30')\n",
    "map_slurm_task_id(sc_name, prefix, 'FASTQ_ONE')\n",
    "map_slurm_task_id(sc_name, srr_id, 'ID')\n",
    "\n",
    "if sps == 'human':\n",
    "    star_cmd(sc_name, human_genome_index, '$FASTQ_ONE', '/nrnb/users/hsher/{}'.format(accession))\n",
    "else:\n",
    "    star_cmd(sc_name, mouse_genome_index, '$FASTQ_ONE', '/nrnb/users/hsher/{}'.format(accession))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "number = 3\n",
    "\n",
    "accession = rnaseq.loc[number, 'GEO accession']\n",
    "srp_id = rnaseq.loc[number, 'SRA']\n",
    "sps = rnaseq.loc[number, 'species']\n",
    "\n",
    "fastq_path = [os.path.join(os.path.join(root_dir, srp_id), file) for file in os.listdir(os.path.join(root_dir, srp_id))]\n",
    "prefix = list(set([srr.split('_')[0] for srr in fastq_path]))\n",
    "srr_id = [p.split('/')[-1].replace('.fastq', '') for p in prefix]\n",
    "\n",
    "sc_name = accession+'_star.run'\n",
    "new_slurm_script(sc_name, '8', '8G', '1-{}'.format(len(srr_id)), '30')\n",
    "map_slurm_task_id(sc_name, prefix, 'FASTQ_ONE')\n",
    "map_slurm_task_id(sc_name, srr_id, 'ID')\n",
    "\n",
    "if sps == 'human':\n",
    "    star_cmd(sc_name, human_genome_index, '$FASTQ_ONE', '/nrnb/users/hsher/{}'.format(accession))\n",
    "else:\n",
    "    star_cmd(sc_name, mouse_genome_index, '$FASTQ_ONE', '/nrnb/users/hsher/{}'.format(accession))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "bam_root = '/nrnb/users/hsher/'\n",
    "gse = rnaseq['GEO accession']\n",
    "sps = rnaseq['species']\n",
    "out_path = bam_root + 'featureCount_out/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "i = 0\n",
    "for g in gse:\n",
    "    script_name = 'featureCount_{0}.run'.format(g)\n",
    "    infile_list = [os.path.join(bam_root+g,file) for file in os.listdir(bam_root+g) if file.endswith(\".bam\")]\n",
    "    outfile = out_path + g\n",
    "    if sps[i] == 'human':\n",
    "        gtf = '/cellar/users/hsher/Data/reference/Homo_sapiens.GRCh38.98.gtf'\n",
    "    else:\n",
    "        gtf = '/cellar/users/hsher/Data/star_mus_musculus_index/Mus_musculus.GRCm38.98.gtf'\n",
    "    \n",
    "    new_slurm_script(script_name, '8', '8G', '1-{}'.format(1), '30')\n",
    "    feature_count(gtf, outfile, infile_list, script_name)\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
